chirp/drivers/ft1d.py

```diff
@@ -1907,66 +1905,37 @@ def _get_digital_settings(self):
-                    c = c + str(s)
+        for i in range(5):
+            cname = f"WiresX_settings.Category[{i}].name"
+            c = str(wxc.Category[i].name).rstrip('\xff').ljust(16)
             val = RadioSettingValueString(0, 16, c)
-            rs = RadioSetting(cname, "Category %d" % (i+1), val)
-            rs.set_apply_callback(self.apply_WiresX_category,
-                                  wxc.Category[i].name)
+            rs = RadioSetting(cname, f"Category {i+1: 2d}", val)
             WXmenu.append(rs)
 
+            WXCmenu = RadioSettingGroup(f"WiresX_settings.Category[{i}]."
+                                        f"RoomsPerCategory",
+                                        f"Cat.[{i + 1}]({c.strip()}).Rooms")
+            WXmenu.append(WXCmenu)
+
             r = wxc.RoomsPerCategory[i]
-            rn = False
-            for j in range(0, 20):
-                idn = "0"
-                if int(r.Rooms[j].ID[1]) != 0xff:
-                    idn = r.Rooms[j].ID
-                    rn = False
-                elif rn:
-                    break
-                elif j > 0:
-                    rn = True
-                val = RadioSettingValueInteger(0, 99999, int(str(idn)))
-                vname = "WiresX_settings.RoomsperCategory%s" \
-                        "Rooms[%d].ID" % (i, j)
-                rs = RadioSetting(vname, "   Room ID%2s (5 numerals)" %
-                                  (j+1), val)
-                rs.set_apply_callback(self.apply_WiresX_roomid,
-                                      r.Rooms[j])
-                WXmenu.append(rs)
-                cn = ''
-                for l in range(0, 16):
-                    s = r.Rooms[j].name[l]
-                    if int(s) != 0xff:
-                        cn = cn + str(s)
-                val = RadioSettingValueString(0, 16, str(cn))
-                cname = "WiresX_settings.RoomsperCategory%s" \
-                        "Rooms[%d].name" % (i, j)
-                rs = RadioSetting(cname, "   Room Name%2s (16 chars)" %
-                                  (j+1), val)
-                rs.set_apply_callback(self.apply_WiresX_roomname,
-                                      r.Rooms[j])
-                WXmenu.append(rs)
+            for j in range(20):
+                cn = str(r.Rooms[j].name).strip('\xff').ljust(16)
+                val = RadioSettingValueString(0, 16, cn)
+                cname = f"WiresX_settings.RoomsPerCategory[{i}]."\
+                    f"Rooms[{j}].name"
+                dname = f"Cat.[{i + 1}] ({c.strip()}) Room{j + 1: 02d}"
+                rs = RadioSetting(cname, dname, val)
+                WXCmenu.append(rs)
+                idn = str(r.Rooms[j].ID).strip('\xff').ljust(5)
+                val = RadioSettingValueString(0, 5, idn)
+                vname = f"WiresX_settings.RoomsPerCategory[{i}]."\
+                    f"Rooms[{j}].ID"
+                rs = RadioSetting(vname, f"   YSF ID{j + 1} (5 digits)",
```

Comment:
Please don't do this sort of hacking of the GUI from the driver. They might look right on your screen, but they likely don't on everyone's and could be entirely not appropriate with a different UI framework. Are you trying to set them off visually? If so, please use `RadioSettingSubGroup()` or something.